app/src/main/java/org/mozilla/focus/navigation/TransactionHelper.java

```diff
@@ -123,24 +169,22 @@ void toggleFakeUrlInput(boolean visible) {
         }
     }
 
-    String getEntryTag(FragmentManager.BackStackEntry entry) {
+    String getFragmentTag(int backStackIndex) {
+        FragmentManager manager = activity.getSupportFragmentManager();
+        return getEntryTag(manager.getBackStackEntryAt(backStackIndex));
+    }
+
+    private String getEntryTag(FragmentManager.BackStackEntry entry) {
         return entryDataSet.get(entry).tag;
     }
 
     private @EntryData.EntryType int getEntryType(FragmentManager.BackStackEntry entry) {
         return entryDataSet.get(entry).type;
     }
 
-    public void updateForegroundType(@EntryData.EntryType int type) {
+    private boolean isStateSaved() {
```

Comment:
You're doing things that are not actually `Rename MainMedaitor to TransactionHelper`. Consider moving these to another commit.

Reply:
done